File: codeforces/31A.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}

	for i := 0; i < n; i++ {
		rem := make(map[int]int)
		for j := 0; j < n; j++ {
			if i == j {
				continue
			}
			needed := a[i] - a[j]
			if idx, found := rem[needed]; found {
				fmt.Printf("%d %d %d\n", i+1, j+1, idx+1)
				return
			}
			rem[a[j]] = j
		}
	}
	fmt.Println(-1)

}

File: codeforces/31A.cpp
#include <bits/stdc++.h>
using namespace std;


int main(){
	int n;
	cin >> n;
	vector<int> a(n);
	for(int i = 0; i < n; i++) {
		cin >> a[i];
	}

	for(int i = 0; i < n; i++) {
		map<int, int> rem;
		for(int j = 0; j < n; j++) {
			if(i == j) continue;
			int needed = a[i] - a[j];
			if (rem.find(needed) != rem.end()) {
				cout << i + 1 << " " << j + 1 << " " << rem[needed] + 1 << endl;
				return 0;
			}
			rem[a[j]] = j;
		}
	}
	cout << -1 << endl;

	return 0;
}
